### PR TITLE
Add cost endpoint and budget manager

### PR DESCRIPTION
**Pull request by yujonglee**
Combine #3 and #4.

### CODE REVIEW

File: tests/test_proxy.py [hunk 1]
```diff
@@ -64,34 +70,75 @@ def test_auth(mock_client, mock_llm, use_valid_api_key):
...
     response = mock_client.get(
-        "/cost/current", headers={"Authorization": "Bearer FASTREPL_INITIAL_KEY"}
+        "/cost/current", headers={"Authorization": f"Bearer {key}"}
     )
     costs = response.json()
     assert costs == {}
+
+
+def test_budget(mock_client, mock_llm, use_valid_api_key):
+    response = mock_client.post(
```
Comment (yujonglee) [inline tests/test_proxy.py:114]: Currently, this test should fail but pass

**Comment by krrishdholakia** [comment]
shouldn't we combine the cost + budget manager? seems like they're both doing the same thing

**Comment by krrishdholakia** [comment]
we should also save the cost / budget to a file - so it doesn't get reset during re-deploys

**Comment by yujonglee** [comment]
@krrishdholakia 
> shouldn't we combine the cost + budget manager? seems like they're both doing the same thing
Yes you are right. 

> we should also save the cost / budget to a file - so it doesn't get reset during re-deploys
Yeah but I think file will not persist, at least for Railway. (https://help.railway.app/usecases/3w6sBTqHudnzM21ehRj45H/mountable-disk-for-storage/tweTazAPQ5pHeu92mecCr7)

We need to store it in Redis than.

**Comment by yujonglee** [comment]
Merge this for now, we will get #7 in and than refactor.

**Pull request merged**